pkg/caching/caching.go

```diff
@@ -0,0 +1,56 @@
+package caching
+
+import (
+	"time"
+
+	"github.com/karlseguin/ccache"
+	"github.com/rancher/opni/pkg/storage"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type InMemoryHttpTtlCache struct {
+	*ccache.Cache
+
+	maxAge time.Duration
+}
+
+func (i InMemoryHttpTtlCache) MaxAge() time.Duration {
+	return i.maxAge
+}
+
+func (i InMemoryHttpTtlCache) Get(key string) (req []byte, ok bool) {
+	item := i.Cache.Get(key)
+	if item == nil || item.Expired() {
+		i.Cache.Delete(key)
+		return nil, false
+	}
+
+	return item.Value().([]byte), true
+}
+
+func (i InMemoryHttpTtlCache) Set(key string, req []byte) {
+	i.Cache.Set(key, req, i.maxAge)
+}
+
+func (i InMemoryHttpTtlCache) Delete(key string) {
+	i.Cache.Delete(key)
```

Comment:
Don't need to override this method as this will already be the behaviour of `i.Delete`

Reply:
It's a little weird but the underlying implementation of the eviction cache has 
```go
// Remove the item from the cache, return true if the item was present, false otherwise.
func (c *Cache) Delete(key string) bool {
```
which doesn't match the interface for the `httpcache` library, which expects nothing